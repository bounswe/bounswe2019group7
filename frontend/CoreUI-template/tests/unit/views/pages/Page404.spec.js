import Vue from 'vue'
import { shallowMount } from '@vue/test-utils'
import BootstrapVue from 'bootstrap-vue'
import Page404 from '@/views/pages/Page404'

Vue.use(BootstrapVue)

describe('Page404.vue', () => {
  it('has a name', () => {
    expect(Page404.name).toMatch('Page404')
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(Page404)
    expect(wrapper.isVueInstance()).toBe(true)
  })
  it('is Page500', () => {
    const wrapper = shallowMount(Page404)
    expect(wrapper.is(Page404)).toBe(true)
  })
  it('should render correct content', () => {
    const wrapper = shallowMount(Page404)
    expect(wrapper.find('h1').text()).toMatch('404')
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(Page404)
    expect(wrapper.element).toMatchSnapshot()
  })
})
