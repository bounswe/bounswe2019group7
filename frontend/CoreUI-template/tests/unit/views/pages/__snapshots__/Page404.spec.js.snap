// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page404.vue renders correctly 1`] = `
<div
  class="app flex-row align-items-center"
>
  <div
    class="container"
  >
    <b-row-stub
      class="justify-content-center"
      tag="div"
    >
      <b-col-stub
        md="6"
        tag="div"
      >
        <div
          class="clearfix"
        >
          <h1
            class="float-left display-3 mr-4"
          >
            404
          </h1>
           
          <h4
            class="pt-3"
          >
            Oops! You're lost.
          </h4>
           
          <p
            class="text-muted"
          >
            The page you are looking for was not found.
          </p>
        </div>
         
        <b-input-group-stub
          tag="div"
        >
          <b-input-group-prepend-stub
            tag="div"
          >
            <b-input-group-text-stub
              tag="div"
            >
              <i
                class="fa fa-search"
              />
            </b-input-group-text-stub>
          </b-input-group-prepend-stub>
           
          <input
            class="form-control"
            id="prependedInput"
            placeholder="What are you looking for?"
            size="16"
            type="text"
          />
           
          <b-input-group-append-stub
            tag="div"
          >
            <b-button-stub
              event="click"
              routertag="a"
              tag="button"
              target="_self"
              type="button"
              variant="info"
            >
              Search
            </b-button-stub>
          </b-input-group-append-stub>
        </b-input-group-stub>
      </b-col-stub>
    </b-row-stub>
  </div>
</div>
`;
